app/controllers/users_controller.rb

```diff
@@ -2,23 +2,37 @@ class UsersController < ApplicationController
   before_action :authenticate_user!
   
   def index
-    @users = User.all
+    @pagy, @users = pagy(User.all)
   end
 
   def edit
-    @users = User.find(params[:id])
+    @user = User.find(params[:id])
+  end
+
+  def update
+    @user = User.find(params[:id])
+  
+    if @user.update(user_params)
+      redirect_to users_path
+    else
+      render "edit"
+    end
   end
   
   def destroy
-    @users = User.find(params[:id])
-    @users.destroy
+    @user = User.find(params[:id])
+    @user.destroy
    
     redirect_to users_path
   end
 
+  def search
+    @pagy, @users = pagy(User.where("first_name Like ? OR last_name Like ?", params[:name].titlecase, params[:name].titlecase))
```

Comment:
refactor into two lines

Reply:
use common variable for params[:name].titlecase

Reply:
done